backtracking/n_queens.cpp

```diff
@@ -1,63 +1,123 @@
+/**
+ * @file n_queens.cpp
+ * @brief [Eight Queens](https://en.wikipedia.org/wiki/Eight_queens_puzzle)
+ * puzzle
+ *
+ * @details
+ * The **eight queens puzzle** is the problem of placing eight chess queens on
+ * an 8×8 chessboard so that no two queens threaten each other; thus, a solution
+ * requires that no two queens share the same row, column, or diagonal. The
+ * eight queens puzzle is an example of the more general **n queens problem** of
+ * placing n non-attacking queens on an n×n chessboard, for which solutions
+ * exist for all natural numbers n with the exception of n = 2 and n = 3.
+ *
+ * @author Unknown author
+ * @author [David Leal](https://github.com/Panquesito7)
+ *
+ */
 #include <iostream>
-#define N 4
-using namespace std;
-
-void printSolution(int board[N][N]) {
-    cout << "\n";
-    for (int i = 0; i < N; i++) {
-        for (int j = 0; j < N; j++) cout << "" << board[i][j];
-        cout << "\n";
+#include <array>
+
+/**
+ * @namespace backtracking
+ * @brief Backtracking algorithms
+ */
+namespace backtracking {
+/**
```

Comment:
can add a member namespace: `n_queens`. This way, the functions here will be separate from other functions in the `backtracking` namespace.

Reply:
Added, thanks.